zerver/actions/streams.py

```diff
@@ -68,6 +71,21 @@
 def do_deactivate_stream(
     stream: Stream, log: bool = True, *, acting_user: Optional[UserProfile]
 ) -> None:
+    # fixes a bug where realm settings are cached in the stream object, but in general it's
+    # a good idea to flush cache and reload before a destructive action like this.
+    flush_realm(instance=stream.realm)
+    flush_stream(instance=stream)
+
+    # Error if this stream is used for reporting messages. Note that this also works for
+    # merge_streams admin command on the command-line.
+    reportmsg_stream = stream.realm.get_reportmsg_stream()
+    if reportmsg_stream is not None and reportmsg_stream.id == stream.id:
```

Comment:
Would it be simpler to just write `if Realm.objects.filter(id=stream.realm_id, reportmsg_stream=stream).exists():`

I think that'd eliminate the need to have the flush calls you have above.

Reply:
DONE - thanks!! good call.